ci-operator/step-registry/gcp/provision/public-dns-zone/api-records/gcp-provision-public-dns-zone-api-records-commands.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
```

Comment:
I'm not sure if we should use GCP to create the expected public DNS, I forgot where I saw it, but I seem to remember that we were supposed to use another DNS provider to do this, like AWS Route53

Reply:
@gpei FYI The step won't create a public DNS zone, instead, it'll create a dns record-set in the existing public DNS zone (which corresponds to the given base domain).

Reply:
yeah, I get it, my doubt is shall we use GCP's cloud DNS to do this for the GCP cluster? or we should change to another Cloud provided public DNS service to host the required DNS.
But one thing I'm not clear now is how we can use other platform's credentials in this GCP job as it's configured with gcp-qe cluster_profile.

Reply:
@gpei the customer would not use GCP's cloud DNS but since its records are created after cluster install is successful, we can be sure it played no part in bringing up the cluster. I think using GCP cloud DNS is a reasonable approximation to what would happen in a customer setup.

Reply:
@sadasu Thanks for the feedback, yes I totally understand this, since the DNS we're going to create later is a public record, it makes no difference to the cluster where it is created. 
We finally chose to create the required DNS records on AWS just to ensure that it would not affect any subsequent E2E tests. The test suites from QE side is a quite large one. In fact, we also don’t know whether there will be any team cases will continue to rely on these DNS on GCP using the previous logic. In order to avoid such problem(maybe only in theory), I discussed with Jianli that we can try to create these DNS on AWS first, if it is difficult to achieve, we will simply create them on GCP. Fortunately, I saw Jianli had already figured out this and automated the process.